ui/src/message_popup/mobile_prototype/text_footer_component.jsx

```diff
@@ -0,0 +1,145 @@
+/* @flow weak */
+import React from 'react';
+
+import Divider from 'material-ui/Divider';
+import TextField from 'material-ui/TextField';
+import RaisedButton from 'material-ui/RaisedButton';
+
+export default React.createClass({
+  displayName: 'TextFooter',
+  
+  propTypes: {
+    onQuestionDone: React.PropTypes.func.isRequired,
+    setInitialResponse: React.PropTypes.func.isRequired,
+    setRevisedResponse: React.PropTypes.func.isRequired,
+    setPassedResponse: React.PropTypes.func.isRequired,
+    log: React.PropTypes.object.isRequired,
+    isReadyToMoveOn: React.PropTypes.bool.isRequired,
+    helpType: React.PropTypes.string.isRequired,
+    elapsedMs: React.PropTypes.number.isRequired,
+    nextButtonLabel: React.PropTypes.string.isRequired,
+    nextExample: React.PropTypes.func.isRequired,
+    onShowExampleClicked: React.PropTypes.func.isRequired,
+    mainStudent: React.PropTypes.object
+  },
+  
+  getInitialState(){
+    return ({
+      text: undefined,
+      initialMessageSent: false,
+    });
+  },
+  
+  onTextChange(event, value){
+    this.setState({text: value});
+  },
+  
+  onSendMessage(){
+    this.props.setInitialResponse(this.state.text);
+    this.setState({initialMessageSent: true});
+    this.props.log.logResponse();
+  },
+  
+  onRevisionSendMessage(){
+    this.props.setRevisedResponse(this.state.text);
+    this.props.log.logRevision();
+  },
+  
+  onPassSendMessage(){
+    this.props.setPassedResponse('');
+    this.props.log.logRevisionDeclined();
+  },
+  
+  onNextQuestion(){
+    this.props.onQuestionDone();
+  },
+  
+  render(){
+    const mainButtonLabel = this.props.isReadyToMoveOn ? this.props.nextButtonLabel : this.props.helpType !== 'feedback' ? 'Respond to ' + (this.props.mainStudent !== undefined ? this.props.mainStudent.name : 'Class') : this.state.initialMessageSent ? 'Revise Response' : 'Save Response';
```

Comment:
This is a long line :)  Can you break this up a bit? Triply nested ternaries are usually harder to read than they're worth and there's a bug with the initial button text in practice mode at the moment that's hard to spot.

I'd suggest splitting this out into a method, and then checking for conditions that you can return early from.  Something like:

```
renderButtonText(): {
  if (this.props.isReadyToMoveOn) return this.props.nextButtonLabel;
  if (this.state.initialMessageSent && this.props.helpType === 'feedback') return 'Revise response';
  ...
}
```